File: ts/examples/quickstart.ts
// Run from ts/: npm run quickstart

import { ContextClipper, type Message } from '../src/index.js'

const clipper = new ContextClipper({ budgetTokens: 200, tailMessages: 2 })

const messages: Message[] = []
for (let i = 0; i < 20; i++) {
  messages.push({
    role: i % 2 === 0 ? 'user' : 'assistant',
    content: `Message number ${i} with some filler content to consume tokens.`,
  })
}

console.log('--- check ---')
console.log(clipper.check(messages))

console.log('\n--- recover ---')
const result = await clipper.recover(messages)
console.log('action:', result.action)
console.log('usedTokens / budgetTokens:', result.usedTokens, '/', result.budgetTokens)
console.log('remaining message count:', result.messages.length, '(from', messages.length, ')')
console.log('first remaining message:', result.messages[0])
console.log('last message (protected tail):', result.messages.at(-1))

File: py/examples/quickstart.py
"""Run from py/: PYTHONPATH=src python3 examples/quickstart.py"""

import asyncio

from contextclip import ContextClipper, Message


async def main() -> None:
    clipper = ContextClipper(budget_tokens=200, tail_messages=2)

    messages = [
        Message(
            role="user" if i % 2 == 0 else "assistant",
            content=f"Message number {i} with some filler content to consume tokens.",
        )
        for i in range(20)
    ]

    print("--- check ---")
    print(clipper.check(messages))

    print("\n--- recover ---")
    result = await clipper.recover(messages)
    print("action:", result.action)
    print("usedTokens / budgetTokens:", result.used_tokens, "/", result.budget_tokens)
    print("remaining message count:", len(result.messages), "(from", len(messages), ")")
    print("first remaining message:", result.messages[0])
    print("last message (protected tail):", result.messages[-1])


if __name__ == "__main__":
    asyncio.run(main())
